cubi_tk/sodar_api.py

```diff
@@ -0,0 +1,151 @@
+        if not self.sodar_url:
+            if not toml_config:
+                raise ParameterException(
+                    "SODAR URL not given on command line and not found in toml config files."
+                )
+            self.sodar_url = toml_config.get("global", {}).get("sodar_server_url")
+            if not self.sodar_url:
+                raise ParameterException(
+                    "SODAR URL not found in config files. Please specify on command line."
+                )
+        if not self.sodar_api_token:
+            if not toml_config:
+                raise ParameterException(
+                    "SODAR API token not given on command line and not found in toml config files."
+                )
+            self.sodar_api_token = toml_config.get("global", {}).get("sodar_api_token")
+            if not self.sodar_api_token:
+                raise ParameterException(
+                    "SODAR API token not found in config files. Please specify on command line."
+                )
+        if not is_uuid(self.project_uuid):
+            logger.warning("Project UUID does not look like valid UUID.")
+
+    @staticmethod
+    def setup_argparse(parser: argparse.ArgumentParser) -> None:
+        """Setup argument parser."""
+        group_sodar = parser.add_argument_group("SODAR-related")
+
+        # load default from toml file
+        # consider: mark token as sensitive
+        group_sodar.add_argument(
+            "--sodar-url",
+            default=os.environ.get("SODAR_URL", "https://sodar.bihealth.org/"),
+            help="URL to SODAR, defaults to SODAR_URL environment variable or fallback to https://sodar.bihealth.org/",
+        )
+        group_sodar.add_argument(
+            "--sodar-api-token",
+            default=os.environ.get("SODAR_API_TOKEN", None),
+            help="Authentication token when talking to SODAR.  Defaults to SODAR_API_TOKEN environment variable.",
+        )
+        group_sodar.add_argument(
+            "project_uuid",
+            help="SODAR project UUID",
+        )
+
+    def _base_api_header(self) -> dict[str, str]:
+        # FIXME: only add versioning header once SODAR API v1.0 is released
+        # (this will introduce individual versioning for specific calls and break the general versioning)
+        sodar_headers = {
+            "Authorization": "token {}".format(self.sodar_api_token),
+            # 'Accept': f'application/vnd.bihealth.sodar+json; version={SODAR_API_VERSION}',
+        }
+        return sodar_headers
+
+    def _api_call(
+        self,
+        api: Literal["samplesheets", "landingzones"],
+        action: str,
+        method: Literal["get", "post"] = "get",
+        params: dict = None,
+        data: dict = None,
+        files: dict = None,
+    ) -> dict:
+        url = self.sodar_url + api + "/api/" + action + "/" + self.project_uuid
+        if params:
+            url += "?" + "&".join([f"{k}={v}" for k, v in params.items()])
```

Comment:
Consider making use of urllib, this will also encode the url properly in case of weird characters etc.